optimade/validator/validator.py

```diff
@@ -825,8 +823,48 @@ def _construct_single_property_filters(
             else:
                 num_data_returned[operator] = response["meta"].get("data_returned")
 
+            if prop in CONF.unique_properties and operator == "=":
```

Comment:
This code block was just moved up from above, with this line being the only change (previously it assumed "=" was already in the dict).